api/src/test/java/com/epam/pipeline/controller/metadata/MetadataControllerTest.java

```diff
@@ -0,0 +1,305 @@
+
+        verify(mockMetadataApiService).deleteMetadataItem(entityVO);
+        assertResponse(mvcResult, metadataEntry, MetadataCreatorUtils.METADATA_ENTRY_TYPE);
+    }
+
+    @Test
+    public void shouldFailDeleteMetadataItem() {
+        performUnauthorizedRequest(delete(DELETE_ITEM_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDeleteMetadataItemKey() throws Exception {
+        final String content = getObjectMapper().writeValueAsString(entityVO);
+        doReturn(metadataEntry).when(mockMetadataApiService).deleteMetadataItemKey(entityVO, TEST_STRING);
+
+        final MvcResult mvcResult = performRequest(delete(DELETE_ITEM_KEY_URL).content(content)
+                .params(multiValueMapOf(KEY, TEST_STRING)));
+
+        verify(mockMetadataApiService).deleteMetadataItemKey(entityVO, TEST_STRING);
+        assertResponse(mvcResult, metadataEntry, MetadataCreatorUtils.METADATA_ENTRY_TYPE);
+    }
+
+    @Test
+    public void shouldFailDeleteMetadataItemKey() {
+        performUnauthorizedRequest(delete(DELETE_ITEM_KEY_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDeleteMetadataItemKeys() throws Exception {
+        final String content = getObjectMapper().writeValueAsString(metadataVO);
+        doReturn(metadataEntry).when(mockMetadataApiService).deleteMetadataItemKeys(metadataVO);
+
+        final MvcResult mvcResult = performRequest(delete(DELETE_ITEM_KEYS_URL).content(content));
+
+        verify(mockMetadataApiService).deleteMetadataItemKeys(metadataVO);
+        assertResponse(mvcResult, metadataEntry, MetadataCreatorUtils.METADATA_ENTRY_TYPE);
+    }
+
+    @Test
+    public void shouldFailDeleteMetadataItemKeys() {
+        performUnauthorizedRequest(delete(DELETE_ITEM_KEYS_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldUploadMetadataFromFile() throws Exception {
+        doReturn(metadataEntry).when(mockMetadataApiService)
+                .uploadMetadataFromFile(eq(new EntityVO(ID, aclClass)), any(), eq(true));
+
+        final MvcResult mvcResult = performRequest(post(UPLOAD_URL).content(MULTIPART_CONTENT)
+                        .params(multiValueMapOf(ENTITY_ID, ID,
+                                                ACL_CLASS, aclClass,
+                                                MERGE_WITH_EXISTING_METADATA, true)),
+                                                MULTIPART_CONTENT_TYPE, EXPECTED_CONTENT_TYPE);
+        final ArgumentCaptor<MultipartFile> multipartFileCaptor = ArgumentCaptor.forClass(MultipartFile.class);
+
+        verify(mockMetadataApiService).uploadMetadataFromFile(any(EntityVO.class),
+                multipartFileCaptor.capture(), eq(true));
+        final MultipartFile returnedFile = multipartFileCaptor.getValue();
+        assertThat(returnedFile.getContentType()).isEqualTo(MediaType.APPLICATION_OCTET_STREAM_VALUE);
+        assertThat(returnedFile.getOriginalFilename()).isEqualTo("file.txt");
```

Comment:
I'm thinking that we should check `returnedFile.getOriginalFilename()` and `file.getBytes()` because these methods used in `MetadataApiService`.

Additionally it seems like a good idea to extract new assertion method which will accept the captor and check for the uploading file name and its content:

```java
assertRequestFile(captor, expectedFileName, expectedContent);
```